framework/graphics/dx12_util.cpp

```diff
@@ -104,14 +105,40 @@ void TakeScreenshot(std::unique_ptr<graphics::DX12ImageRenderer>& image_renderer
 
                             filename += "_frame_";
                             filename += std::to_string(frame_num);
-                            filename += ".bmp";
-
-                            util::imagewriter::WriteBmpImage(filename,
-                                                             static_cast<unsigned int>(fb_desc.Width),
-                                                             static_cast<unsigned int>(fb_desc.Height),
-                                                             datasize,
-                                                             std::data(captured_image.data),
-                                                             static_cast<unsigned int>(pitch));
+
+                            switch (screenshot_format)
+                            {
+                                default:
+                                    GFXRECON_LOG_ERROR(
+                                        "Screenshot format invalid!  Expected BMP or PNG, falling back to BMP.");
+                                    // Intentional fall-through
+                                case gfxrecon::util::ScreenshotFormat::kBmp:
+                                    if (!util::imagewriter::WriteBmpImage(filename + ".bmp",
+                                                                          static_cast<unsigned int>(fb_desc.Width),
+                                                                          static_cast<unsigned int>(fb_desc.Height),
+                                                                          datasize,
+                                                                          std::data(captured_image.data),
+                                                                          static_cast<unsigned int>(pitch)))
+                                    {
+                                        GFXRECON_LOG_ERROR(
+                                            "Screenshot could not be created: failed to write BMP file %s",
+                                            filename.c_str());
+                                    }
+                                    break;
+                                case gfxrecon::util::ScreenshotFormat::kPng:
```

Comment:
Does this need wrapping in `#ifdef GFXRECON_ENABLE_PNG_SCREENSHOT`?

Reply:
That define is not used up in the DX code level.  The `WritePngImage` function call itself though should return false if the #define is not set.  So it will return the error on line 137.